gadgets/trace_sni/program.bpf.c

```diff
@@ -220,9 +220,11 @@ int ig_trace_sni(struct __sk_buff *skb)
 	}
 	event.timestamp_raw = bpf_ktime_get_boot_ns();
 
+	struct sockets_value *skb_val = gadget_socket_lookup(skb);
 	// Enrich event with process metadata
-	gadget_process_populate_from_socket(gadget_socket_lookup(skb),
-					    &event.proc);
+	if (gadget_should_discard_data_by_skb(skb_val))
```

Comment:
Move this check right to the beginning of this function to filter packets early

Reply:
I'm not totally sure this should go to the very beginning as this is a heavy operation and other filtering logic on these gadgets are faster. I moved it a bit above on this gadget and left it on the same position on trace_dns.